src/qibo/models/circuit.py

```diff
@@ -88,24 +85,27 @@ def from_fused(self):
                 queue.append(gate.gates[0])
         return queue
 
-    def append(self, gate: gates.Gate):
-        super().append(gate)
-        if gate.qubits:
-            qubits = gate.qubits
-        else:  # special gate acting on all qubits
-            qubits = tuple(range(self.nqubits))
-
-        if isinstance(gate, gates.M):
-            self.nmeasurements += 1
+    def nmeasurements(self):
```

Comment:
Same here, that can be a `@property` and even `@cached_property` could also work better.